.github/workflows/infracost.yml

```diff
@@ -0,0 +1,165 @@
+        with:
+          python-version: 3.8
+
+      # Install QHub
+      - name: Install QHub
+        if: steps.changes.outputs.${{ matrix.provider }} == 'true'
+        run: |
+          pip install .[dev]
+
+      # Initialize QHub and render the infrastructure
+      - name: QHub initialize and render
+        run: |
+          mkdir data && cd data
+          qhub init "${{ matrix.provider }}" --project "TestProvider" --domain "${{ matrix.provider }}.qhub.dev" --auth-provider github --disable-prompt
+          cat qhub-config.yaml
+          qhub render -c qhub-config.yaml
+
+      # Run Infracost on the base branch to generate the baseline cost
+      - name: Generate Infracost cost estimate baseline
+        run: |
+          infracost breakdown --path=${TF_ROOT} \
+                              --format=json \
+                              --out-file=/tmp/infracost-base.json
+          rm -rf data/
+
+      # Checkout the PR branch
+      - name: Checkout PR branch
+        uses: actions/checkout@v3
+
+      # Install QHub
+      - name: Install QHub
+        if: steps.changes.outputs.${{ matrix.provider }} == 'true'
+        run: |
+          pip install .[dev]
+
+      # Initialize QHub and render the infrastructure
+      - name: QHub initialize and render
+        run: |
+          mkdir data && cd data
+          qhub init "${{ matrix.provider }}" --project "TestProvider" --domain "${{ matrix.provider }}.qhub.dev" --auth-provider github --disable-prompt
+          cat qhub-config.yaml
+          qhub render -c qhub-config.yaml
+
+      # Generate Infracost diff cost estimate only if the file has been changed
+      - name: Generate Infracost diff
+        run: |
+          infracost diff --path=${TF_ROOT} \
+                          --format=json \
+                          --compare-to=/tmp/infracost-base.json \
+                          --out-file=/tmp/infracost.json
+
+      # Generate the cost estimate report and submit it as a PR comment only if the file has been changed
+      - name: Post Infracost comment
+        run: |
+          run_infracost() {
+            infracost comment github --path=/tmp/infracost.json \
+                                      --repo=$GITHUB_REPOSITORY \
+                                      --github-token=${{github.token}} \
+                                      --pull-request=${{github.event.pull_request.number}} \
+                                      --behavior=update
+          }
+          if [[ "${{steps.changes.outputs.aws}}" == "true" ]] && [[ "${{matrix.provider}}" == "aws" ]]; then
+            run_infracost
+          elif [[ "${{steps.changes.outputs.gcp}}" == "true" ]] && [[ "${{matrix.provider}}" == "gcp" ]]; then
+            run_infracost
+          elif [[ "${{steps.changes.outputs.azure}}" == "true" ]] && [[ "${{matrix.provider}}" == "azure" ]]; then
+            run_infracost
+          else
+            echo "No changes detected in infrastructure"
+          fi
```

Comment:
Infracost doesn't support job summaries yet. Let me see if I can make this happen somehow.

Reply:
no rush, I only think it would be cleaner. If infracost has a csv output format we can pass that to a .md file and use that as the summary. -- this is not needed for this PR, it is good already, just was curious about this :slightly_smiling_face:

Reply:
Infracost doesn't seem to generate a job summary yet. I will create an issue for them so that they can look into implementing it 👍

Reply:
Infracost GHA doesn't support job summary yet 😞 . We can look over it if we happen to find a custom job summary feature from them.